cluster-backup-tool.md

```diff
@@ -0,0 +1,144 @@
+Cluster Backup Tool
+====================
+# Requirements
+## Hard
+- Ability to:
+    - Backup data to a file(s) at a given point in time
+    - Recreate data from archive
+- Must be backwards Compatible across x major versions
+- Supported in client/server mode
+
+## Optional/Future
+- Topology aware restoration
+    - Disabled by default, focus is on data
+
+- Import failure modes:
+    - Clusterwide: Rollback all on first error
+    - Per Cache: Only rollback importing of caches which encounter errors
+
+- Cache specific imports
+    - Greater flexibility
+    - Ability to retry failed cache imports
+    - Convert from one cache type to another
+
+- Configuration only backup/import
+
+- Enable "online" backups where the cluster creates a snapshot from a live cluster
+    - Utilise `CommitManager` as used by state transfer and conflict resolution
```

Comment:
If the backup is possible on a "online" cluster the idea is that the `CommitManager` is used to track all of the entries that are put after the backup has been initiated, but before it has completed, as there has to be a cut-off point for when cache entries are accepted as part of the backup. Otherwise in a heavily loaded cache it's possible the backup will never complete.